diff -up luasocket-2.0.2/src/makefile.optflags luasocket-2.0.2/src/makefile
--- luasocket-2.0.2/src/makefile.optflags	2013-05-10 14:00:45.415655745 -0400
+++ luasocket-2.0.2/src/makefile	2013-05-10 14:01:41.625653735 -0400
@@ -117,9 +117,9 @@ CC_linux=gcc
 DEF_linux=-DLUASOCKET_$(DEBUG) -DLUA_COMPAT_MODULE \
 	-DLUASOCKET_API='__attribute__((visibility("default")))' \
 	-DMIME_API='__attribute__((visibility("default")))'
-CFLAGS_linux= -I$(LUAINC) $(DEF) -pedantic -Wall -Wshadow -Wextra -Wimplicit -O2 -ggdb3 -fpic \
+CFLAGS_linux= -I$(LUAINC) $(DEF) $(OPTFLAGS) \
 	-fvisibility=hidden
-LDFLAGS_linux=-O -shared -fpic -o 
+LDFLAGS_linux=$(OPTFLAGS) -shared -o 
 LD_linux=gcc
 SOCKET_linux=usocket.o
 
